### PR TITLE
generalize parseConfFile() so it can handle multiple sections

### PR DESCRIPTION
**Pull request by WalterBright**
This is a start on a fix for https://issues.dlang.org/show_bug.cgi?id=7044

### CODE REVIEW

File: src/mars.c [hunk 1]
```diff
@@ -391,7 +391,15 @@ int tryMain(size_t argc, const char *argv[])
...
-    parseConfFile(global.inifilename, "Environment");
+
+    Strings sections;
+    {
+        /* Read the [Environment] section, so we can later
+         * pick up any DFLAGS settings.
+         */
+        sections.push("Environment");
+        parseConfFile(global.inifilename, &sections);
+    }
```
Comment (yebblies) [inline src/mars.c]: No useless braces please.

**Comment by yebblies** [comment]
What's the plan for the complete fix?

File: src/mars.c (hunk 1 above)
Comment (WalterBright) [inline src/mars.c]: yeah, that was left over from a previous version. removed.

**Comment by WalterBright** [comment]
Be able to select various user-defined sections from the command line.

**Comment by yebblies** [comment]
How will that solve issue 7044?  What would the sections look like?

**Comment by WalterBright** [comment]
Martin explains in the bugzilla issue.

**Comment by yebblies** [comment]
Ah ok, great.

**Comment by yebblies** [comment]
Auto-merge toggled on

**Pull request merged**